toponetx/classes/path_complex.py

```diff
@@ -0,0 +1,657 @@
+"""Path complex."""
+from collections.abc import Hashable, Iterable, Iterator, Sequence
+from itertools import chain
+
+import networkx as nx
+import numpy as np
+import scipy as sp
+
+from toponetx.classes.complex import Complex
+from toponetx.classes.path import Path
+from toponetx.classes.reportviews import NodeView, PathView
+
+__all__ = ["PathComplex"]
+
+
+class PathComplex(Complex):
+    """A class representing a path complex.
+
+    Class for constructing path complexes from graphs or iterables of paths.
+    The path complex is a simplicial complex if certain conditions are met (https://arxiv.org/pdf/1207.2834.pdf).
+
+    Parameters
+    ----------
+    paths : nx.Graph or Iterable[Sequence[Hashable]
+        The paths in the path complex. If a graph is provided, the path complex will be constructed from the graph, and allowed paths are automatically computed.
+    name : str, optional
+        A name for the path complex.
+    reserve_sequence_order : bool, default=False
+        If True, reserve the order of the sub-sequence of nodes in the p-path. Else, the sub-sequence of nodes in the p-path will
+        be reversed if the first index is larger than the last index.
+    allowed_paths : Iterable[tuple[Hashable]], optional
+        An iterable of allowed boundaries. If None, only obvious boundaries are constructed (sub-sequences where the first or the last index is omitted).
+    max_rank : int, default=3
+        The maximal length of a path in the path complex.
+    attr: keyword arguments, optional
+        Additional attributes to be associated with the path complex.
```

Comment:
give a mathematical definition of path complex here. Make it simple and give 2 examples that are easy to understand.

Reply:
Just added.